lib.py

```diff
@@ -54,9 +54,7 @@ async def benchmark(
                     if result is not None:
                         results.append(result)
 
-            if results:
-                with Database() as db:
-                    db.save_results(op_id, year, day, part, code, results)
+                db.save_results(op_id, year, day, part, code, results)
```

Comment:
It doesn't handle None, but the typehints say it doesn't and it passes mypy :D, beauty of typeck, empty list is a bit of a question, since it effectively means the benchmark did nothing, for now it raises an exception in that case, saying it didn't expect for no benchmarks to be passed, but we could change that to a log message too